### ISSUE TITLE
Quake 4 (2210)

### ISSUE THREAD

**Issue by xDShot** [ValveSoftware/Proton#523]
https://store.steampowered.com/app/2210/Quake_IV/

Aside #168, the game works out of the box

**Comment by kisak-valve** [comment]
Hello @xDShot, please copy your system information from steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by xDShot** [comment]
https://gist.github.com/xDShot/072b363fb24811836d146ce3242887ab my system info

**Comment by andrew-mcmahon** [comment]
Quake 4 got ported to Linux by Timothée Besset.
The [installer is still hosted here](https://www.quaddicted.com/files/idgames2/idstuff/quake4/linux/) - you still need the original game files of course. (you don't need the update file)
It might need a bit of [tweaking](https://www.gamingonlinux.com/articles/playing-quake-4-on-linux-in-2018.11017) to get it going.

[Prey](https://icculus.org/prey/) works as well with some [changes](https://www.gamingonlinux.com/articles/playing-prey-on-linux-in-2018.11118). I've got that working.

**Comment by dreamer** [comment]
@andrew-mcmahon installer for Q4, that you linked is not available for EU users, for no apparent reason.

> Access from European IPs blocked due to the legal danger of GDPR... :(
> 
> Please see https://www.quaddicted.com/forum/viewtopic.php?id=578
> if you can help with *definitive*, *concrete* answers to GDPR compliance.

Apparently, owners have no idea what GDPR is. I would answer in that forum, except forum link is also blocked.

**Comment by andrew-mcmahon** [comment]
> @andrew-mcmahon installer for Q4, that you linked is not available for EU users, for no apparent reason.
> 
> > Access from European IPs blocked due to the legal danger of GDPR... :(
> > Please see https://www.quaddicted.com/forum/viewtopic.php?id=578
> > if you can help with _definitive_, _concrete_ answers to GDPR compliance.
> 
> Apparently, owners have no idea what GDPR is. I would answer in that forum, except forum link is also blocked.

Strange as it certainly isn't warez or anything funny, this was an official release. 
You can try reading the forum link by going through the [archive.org time machine](https://web.archive.org/web/20180818110945/https://www.quaddicted.com/forum/viewtopic.php?id=578).
GDPR is the privacy law what got brought in isn't it? It's why I'm getting annoying cookie popups on every website I visit ='(
Our EU overlords haven't blocked the link in the UK.

The filename to google is: **quake4-linux-1.4.2.x86.run**
**sha1sum:** 870bbda920b846c51904b22c40ce15aac8c9183b
**md5sum:** bd2cb8164ede4272adcc091f724855ac

There's other links [here](http://distro.ibiblio.org/rootlinux/rootlinux-ports/more/quake4/) and [here](http://ftp.vim.org/ibiblio/distributions/rootlinux/ports/more/quake4/) and possibly a few other places.
You need the original game files (.pk4) which is all explained in the included readme. 
STEAM or GOG (i.e. innoextract setup_quake_4_1.4.3_(13519).exe --gog) for those.

**Comment by andrew-mcmahon** [comment]
Here's a few [screenshots](https://imgur.com/a/JZ3lQ0X)

In case it isn't clear you need to run the installer installing Quake4 into an appropriate place i.e. **/home/user/quake4**
Annoyingly it insists on making symlinks in **/usr/local/bin**. fakeroot maybe? 
I just used sudo on the installer and changed the file permissions afterwords.

After that's done you need to copy the required pk4 files into the **quake4/q4base folder**
You may run into the [serial key issue](https://www.gog.com/forum/quake_series/quake_4_on_linux_no_cd_key/post31) - you can redirect the check by editing /etc/hosts i.e:
127.0.0.1       q4master.idsoftware.com

Other than that the game appears to be fine. 
Be nice if Valve could flex their muscles and "convince" Bethesda to release this for STEAMOS with some alterations.
The source code for _Quake 4_ and _Prey_ was never released unlike _Doom 3_

**Comment by xDShot** [comment]
Linux port is based on old SDL1 that doesn't threat several keyboard buttons like switching system volume or alt+tab, and it is not compatible with mods with custom game binary code. And gamesaves are not cross-platform compatible. Not worthing.

**Comment by dreamer** [comment]
> GDPR is the privacy law what got brought in isn't it? It's why I'm getting annoying cookie popups on every website I visit ='(

No, GDPR is a law mandating companies to inform you, if they are selling your data. Cookie notifications are not part of it, just install ["I don't care about cookies"](https://addons.mozilla.org/pl/firefox/addon/i-dont-care-about-cookies/) to get rid of them :)

> Our EU overlords haven't blocked the link in the UK.

EU didn't block anything. Owner of quaddicted.com blocked access to EU users for no reason.

Anyway, thank you for links to linux packages :) **Let's keep this issue about Proton-specific problems from now on.**

**Comment by lucifertdark** [comment]
Actually they blocked

> > GDPR is the privacy law what got brought in isn't it? It's why I'm getting annoying cookie popups on every website I visit ='(
> 
> No, GDPR is a law mandating companies to inform you, if they are selling your data. Cookie notifications are not part of it, just install ["I don't care about cookies"](https://addons.mozilla.org/pl/firefox/addon/i-dont-care-about-cookies/) to get rid of them :)
> 
> > Our EU overlords haven't blocked the link in the UK.
> 
> EU didn't block anything. Owner of quaddicted.com blocked access to EU users for no reason.
> 
> Anyway, thank you for links to linux packages :) **Let's keep this issue about Proton-specific problems from now on.**

Actually they blocked access to EU users because it's easier to do that than face the hefty fines because they're not complying with the law. AND the cookie notifications ARE part of it.

Under the new legislation, websites need to get explicit permission for each use of data and companies must be more transparent about how they store and use personal data. In the instance of a data breach, the company must inform any users affected by it within 72 hours. Companies that fail to comply face weighty fines or forfeiting up to 4% of their global revenue.

**Comment by dreamer** [comment]
> Actually they blocked access to EU users because it's easier to do that than face the hefty fines because they're not complying with the law.

Didn't know quaddicted.com is a company employing 250 or more people? They probably are not, therefore they don't need to be afraid of hefty fines - they still need to inform users about their privacy policy, etc, but *that should not be a problem*.

**Comment by shoober420** [comment]
# Whitelist Request
- Name of the game to be whitelisted: Quake 4
- Steam AppID of the game: https://steamdb.info/app/2210/

## System Information
- GPU: GTX 760
- Driver/LLVM version: NVIDIA 440.44
- Distro version: Arch
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfonvidia
- Proton version: 4.11-11

Works great out of the box, but a bug thats present even on the Windows version, is carried over to Linux. Its the low texture resolution bug, causes by Quake 4 not correctly calculating how much video RAM you have. Details about this and how to fix is linked below.

https://www.pcgamingwiki.com/wiki/Quake_4#Poor_image_quality_on_modern_systems

**Comment by wantija** [comment]
tried proton 8.04  and experimental on my desktop (mesa-git, rx 6800xt, linux 6.5) and the game renders to a black screen, you can hear the intro videos play and the mouse hover sound in the main menu. The game works on Wine 8.15 staging. Log is attached.
[steam-2210.log](https://github.com/ValveSoftware/Proton/files/12876729/steam-2210.log)

**Comment by LethalManBoob** [comment]
[steam-2210.log](https://github.com/user-attachments/files/26312445/steam-2210.log)

quake 4 no longer boots on stable proton 10. I CAN sometimes get it to boot on proton 9 but changing config settings always borks it

**Comment by kisak-valve** [comment]
Hello @LethalManBoob, `warn:seh:handle_syscall_fault backtrace: --- Exception 0xc0000005 at 0xdef0a1a0: /run/host/usr/lib32/libvulkan_radeon.so + 0x31e1a0.` hints towards a video driver issue that should also be mentioned to your video driver vendor if it hasn't been already.

**Comment by LethalManBoob** [comment]
I can confirm that mesa-git fixes this, so i will report
